packages/core-flows/src/definition/cart/workflows/create-carts.ts

```diff
@@ -51,11 +83,53 @@ export const createCartWorkflow = createWorkflow(
       }
     )
 
-    // TODO: Add line items
+    const variants = getVariantsStep({
+      filter: { id: variantIds },
+      config: {
+        select: [
+          "id",
+          "title",
+          "sku",
+          "barcode",
+          "product.id",
+          "product.title",
+          "product.description",
+          "product.subtitle",
+          "product.thumbnail",
+          "product.type",
+          "product.collection",
+          "product.handle",
+        ],
+        relations: ["product"],
+      },
+    })
+
+    const lineItems = transform({ priceSets, input, variants }, (data) => {
+      const items = (data.input.items ?? []).map((item) => {
+        const variant = data.variants.find((v) => v.id === item.variant_id)!
+
+        return prepareLineItemData({
+          variant: variant,
+          unitPrice: data.priceSets[item.variant_id].calculated_amount,
+          quantity: item.quantity,
+          metadata: item?.metadata ?? {},
+        })
+      })
+
+      return items
+    })
+
+    const cartToCreate = transform({ lineItems, cartInput }, (data) => {
+      return {
+        ...data.cartInput,
+        items: data.lineItems,
+      }
+    })
+
+    const carts = createCartsStep([cartToCreate])
 
-    // @ts-ignore
-    const cart = createCartsStep([cartInput])
+    const cart = transform({ carts }, (data) => data.carts?.[0])
```

Comment:
With the new type changes, we should be able to do just this:

```suggestion
    return carts[0]
```

Reply:
Are you sure? This doesn't work for me. I am seeing the following error:

![CleanShot 2024-02-26 at 13 57 19](https://github.com/medusajs/medusa/assets/59018053/ddf48119-40ac-4949-a31c-2f433c874e9a)


And I believe this is because we want to access an element of an unresolved variable. If we were just returning `carts` then it would be a different case, but I am not sure.

Reply:
ah no, you're right there.